silverback/application.py

```diff
@@ -73,29 +73,33 @@ def __init__(self, settings: Optional[Settings] = None):
             f"{signer_str}{start_block_str}{new_block_timeout_str}"
         )
 
+    def checkpoint(self, last_block_seen: int, last_block_processed: int):
+        self.broker.state.last_block_seen = last_block_seen
+        self.broker.state.last_block_processed = last_block_processed
+
     def on_startup(self) -> Callable:
         """
-        Code to execute on worker startup / restart after an error.
+        Code to execute on client startup / restart after an error.
 
         Usage example::
 
             @app.on_startup()
             def do_something_on_startup(state):
                 ...  # Can provision resources, or add things to `state`.
         """
-        return self.broker.on_event(TaskiqEvents.WORKER_STARTUP)
+        return self.broker.on_event(TaskiqEvents.CLIENT_STARTUP)
```

Comment:
I'm not convinced this change is appropriate.  It kind of made sense when thinking about early processing, but this isn't exactly key to this PR.

On a single in-memory broker it's pretty much the same.  I'm wondering if silverback needs to support both client and worker startup events.

Reply:
Woke up with a change of mind.  I added on_client_* and on_worker_* Silverback decorators, and changed our `on_startup()` decorator to be for a single fired task at runner startup.  This way we don't have an issue of the client doing work, and we don't have multiple workers duplicating work.  It's a job that should run once on one worker.

Reply:
I'm not sure exactly what client startup and shutdown events are useful for.  Should consider if this is useful to users and if not, remove.

Reply:
I think they use it to initialize dependencies into worker state, such as a DB connection that the worker uses

Reply:
I think our use case is more specific to the client side, as our client just manages the subscription queue and with persistence we want to fetch and store the results

Reply:
When I say "client" I mean the runner side.  It can't init anything in worker mem (InMemoryBroker notwithstanding).  I get why worker startup handlers are useful.  Just not picturing the runner side.

I don't think client-side handlers work for our use case either.  We currently use it in a way that acts as a job handler.  Which is why I added the new internal startup/shutdown jobs.

Reply:
So I confirmed this in my testing of distributed Silverback today.  As this PR is right now:

- `@app.on_client_startup()`: Executes on client (runner) side upon startup before jobs are scheduled. This can alter the client's state.  State is unique to the process.  It is not available to workers.  Unclear what use case this might have.
- `@app.on_worker_startup()`: Executes on every worker upon startup.  Can setup taskiq state and DB connections for that worker.
- `@app.on_startup()`: is a task executed **once** on one worker.  This is what is useful for the ApePay example when processing historical events missed.

Reply:
I'm going to leave in `on_client_startup()` unless there's any objections.  I think it's a probably a foot-gun, so maybe we just leave it undocumented.